src/test/java/reposense/util/InputBuilder.java

```diff
@@ -0,0 +1,122 @@
+package reposense.util;
+
+import java.nio.file.Path;
+
+import reposense.parser.ArgsParser;
+
+/**
+ * A utility class to help with building command line input.
+ * Example usage: <br>
+ *     {@code String input = new InputBuilder().setSinceDate("27/01/2017").build();}
+ */
+public class InputBuilder {
+    private static final String WHITESPACE = " ";
+
+    private String help;
+    private String config;
+    private String repos;
+    private String view;
+    private String output;
+    private String sinceDate;
+    private String untilDate;
+    private String formats;
+    private String ignoreStandaloneConfig;
+    private String extraString;
```

Comment:
These changes are not necessary.
Reason: Test util shouldn't interfere with the parsing, e.g. double params such as `--config ... --config ... ` because this is handled by the ArgParser.

Reply:
Also use a `StringBuilder` instead. It is more efficient than concatenation in loops.

Reply:
> These changes are not necessary.
> Reason: Test util shouldn't interfere with the parsing, e.g. double params such as `--config ... --config ... ` because this is handled by the ArgParser.

@yamidark So I revert the latest commit?

Reply:
>These changes are not necessary.
Reason: Test util shouldn't interfere with the parsing, e.g. double params such as `--config ... --config ... ` because this is handled by the ArgParser.

@eugenepeh From our discussion [here](https://github.com/reposense/RepoSense/pull/526#discussion_r256039060), from testing by @yong24s, using double params or multiple params gives undefined behaviour, so I thought it would be good to ensure that we only use single params for testing though.

Reply:
continuing the discussion from that thread:

>@yong24s Used 2 `-config` params and it took the 1st one, while when I used 2 `-config` params it took the last one instead, which is why we thought the behaviour is undefined/unpredictable depending on how to user uses it

Hmm I would prefer keep it as similar as the terminal input as possible. The bad thing about this change is that it keeps the params in a rigid order and can cause `extraString` to be treated as input of any param. 

If there's an issue with the `undefined/unpredictable` input, we can just avoid testing multiple same param as a whole; since we have environment for 3 different os, it should have low possibly of passing the tests if thats the case.

Reply:
>we can just avoid testing multiple same param as a whole

Well, I asked for this change to prevent 'accidentally' setting up any tests for this (e.g. using the same `InputBuilder` for 2 tests within a single test method) 😅. But then again, I think this can be prevented by making `build` to call `reset` after building everything.

>The bad thing about this change is that it keeps the params in a rigid order and can cause `extraString` to be treated as input of any param.

Hmmm, thats true as well, params must be in a specific order (`-help` comes first, than `-config` and so on) which prevents certain tests as well. I guess we can revert to the old method, but add Javadocs to warn developers that they should only be called once for each `build`?

Reply:
>Hmmm, thats true as well, params must be in a specific order (-help comes first, than -config and so on) which prevents certain tests as well. I guess we can revert to the old method, but add Javadocs to warn developers that they should only be called once for each build?

Yeah, javadocs should be good enough.

Reply:
@fzdy1914 Could you revert back to the old version, this time add Javadocs as mentioned. Sorry for the trouble caused.